src/main/java/org/opensrp/repository/postgres/SettingRepositoryImpl.java

```diff
@@ -520,7 +533,68 @@ private void checkWhetherMetadataExistsBeforeSave(SettingConfiguration entity, L
 				settingsMetadataList.add(metadata);
 			}
 		}
-			settingMetadataMapper.insertMany(settingsMetadataList);
+
+		insertSettingMetadata(settingsMetadataList);
+	}
+
+	private void insertSettingMetadata(List<SettingsMetadata> settingsMetadataList) {
+		try {
+			String insertSettingMetadata = "INSERT INTO core.settings_metadata ( settings_id, document_id, identifier, "
+					+ "server_version, team, team_id, provider_id, location_id, uuid, json, setting_type, setting_value, setting_key, setting_description, setting_label, inherited_from) VALUES(?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?)";
+
+			Connection connection = null;
+
+			try {
+				connection = DataSourceUtils.getConnection(openSRPDataSource);
+				PreparedStatement preparedStatement = connection.prepareStatement(insertSettingMetadata);
+				final int batchSize = 1000;
+				int count = 0;
+				for (SettingsMetadata settingsMetadata : settingsMetadataList) {
+
+					ObjectMapper objectMapper = new ObjectMapper();
+					String json = objectMapper.writeValueAsString(settingsMetadata.getJson());
+
+					PGobject pGobject = new PGobject();
+					pGobject.setType("json");
+					pGobject.setValue(json);
+
+					preparedStatement.setLong(1, settingsMetadata.getSettingsId());
+					preparedStatement.setString(2, settingsMetadata.getDocumentId());
+					preparedStatement.setString(3, settingsMetadata.getIdentifier());
+					preparedStatement.setLong(4, settingsMetadata.getServerVersion());
+					preparedStatement.setString(5, settingsMetadata.getTeam());
+					preparedStatement.setString(6, settingsMetadata.getTeamId());
+					preparedStatement.setString(7, settingsMetadata.getProviderId());
+					preparedStatement.setString(8, settingsMetadata.getLocationId());
+					preparedStatement.setString(9, settingsMetadata.getUuid());
+					preparedStatement.setObject(10, pGobject);
+					preparedStatement.setString(11, settingsMetadata.getSettingType());
+					preparedStatement.setString(12, settingsMetadata.getSettingValue());
+					preparedStatement.setString(13, settingsMetadata.getSettingKey());
+					preparedStatement.setString(14, settingsMetadata.getSettingDescription());
+					preparedStatement.setString(15, settingsMetadata.getSettingLabel());
+					preparedStatement.setString(16, settingsMetadata.getInheritedFrom());
+					preparedStatement.addBatch();
+					if (++count % batchSize == 0) {
+						preparedStatement.executeBatch();
+					}
+				}
+				preparedStatement.executeBatch(); // insert remaining records
+				preparedStatement.close();
+				connection.close();
+			}
+			finally {
+				if (connection != null) {
+					DataSourceUtils.releaseConnection(connection, openSRPDataSource);
+				}
+			}
+		}
+		catch (SQLException e) {
+			logger.error(e.getMessage(), e);
+		}
+		catch (JsonProcessingException e) {
+			logger.error(e.getMessage(), e);
+		}
```

Comment:
since code supports JDK 8 and above simplify to `catch (SQLException|JsonProcessingException e)` since errors are just logged. Also should you maybe throw an exception or not catch them so that a this error is not hidden and API can return an error